src/main/scala/com/tsunderebug/scolor/otf/types/OTFEncodingRecord.scala

```diff
@@ -82,7 +83,16 @@ object OTFEncodingRecord {
 
 }
 
-case class SequentialMapGroup(startCodepoint: Codepoint, endCodepoint: Codepoint, startGlyphID: GlyphID) extends EnclosingSectionDataType {
+case class SequentialMapGroup(startCodepoint: Codepoint, endCodepoint: Codepoint, startGlyphID: GlyphID) extends EnclosingSectionDataType with Ordered[SequentialMapGroup] {
+
+  /** Compare this Group to another
+   * @param that The SequentialMapGroup to compare this instance to
+   * @note Groups must be sorted by increasing startCharCode. A group's endCharCode must be less than the startCharCode of the following group, if any. The endCharCode is used, rather than a count, because comparisons for group matching are usually done on an existing character code, and having the endCharCode be there explicitly saves the necessity of an addition per group.
+   * @see https://www.microsoft.com/typography/otspec/cmap.htm#format12
+   */
+  def compare(that: SequentialMapGroup) = {
```

Comment:
Sorting should be used in the getBytes of SegmentedCoverage

Reply:
Ok, so we'll need to override the getBytes method in this class since we inherit the one from `EnclosingSectionDataType` then

Reply:
Sorry, I mean order in `sections` @EdgeCaseBerg as the getBytes gets data from there.

Reply:
Little confused. Let me type it out so you can help me see where to sort those groups. 

![Spec Screenshot](https://my.mixtape.moe/glsskr.jpg)

From what I can gather with the comment I referenced and the spec itself the parameter to the constructor of `SegmentedCoverageEncodingFormat` is the groups themselves which are expected to be ordered right? But if a user access's the `smg` field they won't be. We can work around that like so:

```
  case class SegmentedCoverageEncodingFormat(private val smg: Seq[SequentialMapGroup]) extends EncodingFormat {

    val smg = smg.sorted //We could rename this to groups to more closely match the spec too
```

which would make `smg` no longer a property of `SegmentedCoverageEncodingFormat` in the usual sense. you only really need to make the constructor argument a private val if they have the same name for the field in the class body I believe, so if one was named `smg` and the other `groups` there wouldn't be a conflict though then people could access the unsorted `smg` and the sorted `groups` which might be kinda weird.

I'm not entirely following what to do about the `sections` since the `SegmentedCoverageEncodingFormat` has

```
    override def sections(b: ByteAllocator): Seq[Section] = Seq(
      Section("format", OTFUInt16(UShort(12))),
      Section("reserved", OTFUInt16(UShort(0))),
      Section("length", OTFUInt32(length(b)))
    )
```

which doesn't have a "groups" section or anything like that. Am I looking in the wrong place?

Reply:
Yeah. Where `SegmentedCoverageEncodingFormat` defines `sections`, sort the `Seq[SequentialMapGroup]` in the `Section` case class.

Reply:
I'm still not sure if I'm following exactly since the `sections` don't contain any of the groups yet, but I'm interpreting your comment as saying that they should be added. So then the code would end up like:

```
    override def sections(b: ByteAllocator): Seq[Section] = Seq(
      Section("format", OTFUInt16(UShort(12))),
      Section("reserved", OTFUInt16(UShort(0))),
      Section("length", OTFUInt32(length(b))),
      Section("numGroups", OTFUInt32(UInt(smg.length))),
    ) ++ smg.sorted.zipWithIndex.map {
        case (group, index) => Section(s"groups[${index}]", group)
      }
```

Which would place the `SequentialMapGroup` sections in the section data, sorted, at the end. Also, when the spec says that the name of the group section array is `groups[numGroups]` does it literally mean that? Or is it more like what I have above where there's an actual index replacing `numGroups`? If it mentioned this on the page I must have missed it

Reply:
If it is what I'm describing above then I can add that in as well as a test against the sections section to confirm they're ordered and whatnot

Reply:
UM I THINK I MADE A MISTAKE LEMME LOOK

Yeah, there's supposed to be a section there for the records:

```
    override def sections(b: ByteAllocator): Seq[Section] = Seq(
      Section("format", OTFUInt16(UShort(12))),
      Section("reserved", OTFUInt16(UShort(0))),
      Section("length", OTFUInt32(length(b))),
      Section("numGroups", OTFUInt32(UInt(smg.length))),
      Section("records", OTFArray(smg.sorted.toArray))
    )
```

Reply:
Ah! I forgot about the `OTFArray`!